File: app/server/notify/channels/weplusbot.ts
/**
 * 微加机器人推送渠道
 */
import Logger from '../../utils/logger';
import { NotifyChannel, NotifyResult } from '../types';
import { $ } from '../httpClient';
import { hasConfig, getConfig } from '../config';

const logger = Logger.child({ channel: 'weplusbot' });

export const weplusbotChannel: NotifyChannel = {
    name: 'weplusbot',
    enabled: false,

    async send(text: string, desp: string): Promise<NotifyResult> {
        return new Promise((resolve) => {
            if (!hasConfig('WE_PLUS_BOT_TOKEN')) {
                resolve({ success: false, message: 'WE_PLUS_BOT_TOKEN 未配置' });
                return;
            }

            const WE_PLUS_BOT_TOKEN = getConfig('WE_PLUS_BOT_TOKEN');
            const WE_PLUS_BOT_RECEIVER = getConfig('WE_PLUS_BOT_RECEIVER');
            const WE_PLUS_BOT_VERSION = getConfig('WE_PLUS_BOT_VERSION') || 'pro';

            const url = WE_PLUS_BOT_VERSION === 'personal'
                ? 'https://personal.weplusbot.com/send'
                : 'https://pro.weplusbot.com/send';

            const body = {
                token: WE_PLUS_BOT_TOKEN,
                receiver: WE_PLUS_BOT_RECEIVER,
                title: text,
                content: desp
            };

            $.post({ url, json: body }, (err, _resp, data) => {
                if (err) {
                    logger.error({ err }, '微加机器人发送失败');
                    resolve({ success: false, error: err as Error });
                } else {
                    const result = data as { code?: number; message?: string };
                    if (result.code === 0 || result.code === 200) {
                        logger.info('微加机器人发送成功');
                        resolve({ success: true, message: '发送成功' });
                    } else {
                        logger.warn({ body: data }, '微加机器人发送失败');
                        resolve({ success: false, message: result.message || '发送失败' });
                    }
                }
            });
        });
    }
};

if (hasConfig('WE_PLUS_BOT_TOKEN')) {
    weplusbotChannel.enabled = true;
}

export default weplusbotChannel;

File: app/server/internal/notify/channels/weplusbot.go
package channels

import (
	"log/slog"

	"github.com/sushazhi/fnos-logmanager/internal/notify"
)

type WePlusBot struct {
	enabled bool
}

func (c *WePlusBot) Name() string {
	return "weplusbot"
}

func (c *WePlusBot) Enabled() bool {
	return c.enabled
}

func (c *WePlusBot) SetEnabled(enabled bool) {
	c.enabled = enabled
}

func (c *WePlusBot) Send(text, desp string) notify.NotifyResult {
	if !notify.HasConfig("WE_PLUS_BOT_TOKEN") {
		return notify.NotifyResult{Success: false, Message: "WE_PLUS_BOT_TOKEN 未配置"}
	}

	wePlusBotToken := notify.GetConfig("WE_PLUS_BOT_TOKEN")
	wePlusBotReceiver := notify.GetConfig("WE_PLUS_BOT_RECEIVER")
	wePlusBotVersion := notify.GetConfig("WE_PLUS_BOT_VERSION")
	if wePlusBotVersion == "" {
		wePlusBotVersion = "pro"
	}

	var url string
	if wePlusBotVersion == "personal" {
		url = "https://personal.weplusbot.com/send"
	} else {
		url = "https://pro.weplusbot.com/send"
	}

	body := map[string]string{
		"token":    wePlusBotToken,
		"receiver": wePlusBotReceiver,
		"title":    text,
		"content":  desp,
	}

	resp, err := notify.HTTPPost(url, notify.HttpRequestOptions{
		JSON: body,
	})
	if err != nil {
		slog.Error("微加机器人发送失败", "error", err)
		return notify.NotifyResult{Success: false, Message: err.Error(), Err: err}
	}

	if resp.StatusCode == 200 {
		slog.Info("微加机器人发送成功")
		return notify.NotifyResult{Success: true, Message: "发送成功"}
	}

	slog.Warn("微加机器人发送失败", "status", resp.StatusCode, "body", resp.Body)
	return notify.NotifyResult{Success: false, Message: "发送失败"}
}

func init() {
	ch := &WePlusBot{}
	if notify.HasConfig("WE_PLUS_BOT_TOKEN") {
		ch.SetEnabled(true)
	}
	notify.Registry.Register(ch)
}
